### DISCUSSION TITLE
How to apply plot bounds?

### DISCUSSION THREAD

**Discussion by Peepo-Juice** [emilk/egui#2622] [Q&A]
So for some reason, when I set the plot bounds for a Plot, it does not apply until I click on the plot. I think it has something to do with the drag event, because if I disable that, it becomes even harder to get it to apply the bounds. I have to use the box zoom to get it to apply.

Feels like a bug to me, in which case I should report it.

Oh i should clarify that the order of events go like this: while data array is empty, the plot is not drawn. When data is in the array, plot will start drawing and plot bounds is getting set at this point. Visually, the plot is now at the point where its showing all the data (So the plot bounds is not being applied) as if i had double clicked the plot to reset it. It will remain like this, ignoring the plot bounds, until I click on it (if dragging is enabled).

**Comment by c-git** [comment]
Maybe this was fixed as this was a while ago but in my application I think it works ok for me. The application I'm working on is not Open Source but the version I presented last month at a meetup is online [here](https://github.com/vancouver-rs/talks/tree/master/2023-12-20-adventures-in-egui-app-dev/manual_data_creator). I set the plot bounds in the reset button at the bottom because of how I want reset to work.

**Reply by Peepo-Juice** [reply]
Wow time flies!

I'm not sure if I can easily test this again, but if I do and get any new info, I'll post it here.

**Reply by c-git** [reply]
Ok no worries, let me know if you run into any issues.
